service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/ManageOrdersAboutToSubmitControllerTest.java

```diff
@@ -118,6 +136,46 @@ void shouldBuildNewOrderObject() {
         );
     }
 
+    @Test
+    void shouldBuildNewEPOObject() {
+        final ManageOrdersEventData manageOrdersEventData = buildManageOrdersEventData();
+        CaseData caseData = buildCaseData().toBuilder().manageOrdersEventData(manageOrdersEventData).build();
+
+        CaseData responseCaseData = extractCaseData(postAboutToSubmitEvent(caseData));
+
+        assertThat(responseCaseData.getOrderCollection()).containsOnly(
+            element(ELEMENT_ID, GeneratedOrder.builder()
+                .orderType("C23_EMERGENCY_PROTECTION_ORDER")
+                .title("C23 - Emergency protection order")
+                .children(CHILDREN)
+                .judgeAndLegalAdvisor(JudgeAndLegalAdvisor.builder()
+                    .judgeTitle(HIS_HONOUR_JUDGE)
+                    .judgeLastName("Dredd")
+                    .build())
+                .dateTimeIssued(now())
+                .approvalDateTime(manageOrdersEventData.getManageOrdersApprovalDateTime())
+                .childrenDescription("first1 last1, first2 last2")
+                .document(DOCUMENT_PDF_REFERENCE)
+                .unsealedDocumentCopy(DOCUMENT_WORD_REFERENCE)
+                .build()));
+    }
+
+    private ManageOrdersEventData buildManageOrdersEventData() {
```

Comment:
I would make this local to the method as it is very specific event data

Reply:
What you could do instead is also extract the event data from `buildCaseData()` and make that local to the other test. As such that method is now building a generic case data for you to then add to.